File: app/src/main/java/gardezi/io/amiibofinder/rx/RxAndroidViewModel.java
package gardezi.io.amiibofinder.rx;

import android.arch.lifecycle.ViewModel;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class RxAndroidViewModel extends ViewModel {

    private final CompositeDisposable disposables = new CompositeDisposable();

    public RxAndroidViewModel() {

    }

    protected void addDisposable(Disposable disposable) {
        this.disposables.add(disposable);
    }

    protected void clearDisposables() {
        this.disposables.clear();
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        disposables.clear();
    }
}

File: app/src/main/java/gardezi/io/amiibofinder/rx/RxAndroidViewModel.kt
package gardezi.io.amiibofinder.rx

import android.arch.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

abstract class RxAndroidViewModel : ViewModel() {

    private val disposables = CompositeDisposable()

    protected fun addDisposable(disposable: Disposable) {
        this.disposables.add(disposable)
    }

    protected fun clearDisposables() {
        this.disposables.clear()
    }

    override fun onCleared() {
        super.onCleared()
        disposables.clear()
    }
}
